docker/server/configs/nginx.conf

```diff
@@ -1,31 +1,76 @@
-user www-data;
+# /etc/nginx/nginx.conf
+
+user nginx;
+
+# Set number of worker processes automatically based on number of CPU cores.
 worker_processes 2;
-daemon off;
+
+# Enables the use of JIT for regular expressions to speed-up their processing.
+pcre_jit on;
+
+# Configures default error logger.
+# access_log /dev/stdout nginx;
+# error_log /dev/stderr warn;
+
+# Includes files with directives to load dynamic modules.
+include /etc/nginx/modules/*.conf;
 
 events {
+    # The maximum number of simultaneous connections that can be opened by
+    # a worker process.
     worker_connections 5000;
     multi_accept on;
 }
 
 worker_rlimit_nofile 20000;
 
 http {
-    sendfile off;
-    tcp_nopush on;
-    tcp_nodelay on;
-    keepalive_requests 100;
-    keepalive_timeout 30;
+    # Includes mapping of file name extensions to MIME types of responses
+    # and defines the default type.
+    include /etc/nginx/mime.types;
+    default_type application/octet-stream;
     types_hash_max_size 2048;
+
+    # Name servers used to resolve names of upstream servers into addresses.
+    # It's also needed when using tcpsocket and udpsocket in Lua modules.
+    #resolver 208.67.222.222 208.67.220.220;
+
+    # Don't tell nginx version to clients.
     server_tokens off;
-    client_max_body_size 128M;
 
-    include /etc/nginx/mime.types;
-    default_type text/plain;
+    # Specifies the maximum accepted body size of a client request, as
+    # indicated by the request header Content-Length. If the stated content
+    # length is greater than this size, then the client receives the HTTP
+    # error code 413. Set to 0 to disable.
+    client_max_body_size 300m;
```

Comment:
Looks like it is better to use common default.conf for both client and server? This can then be added in common config

Reply:
Actually same nginx.conf can be shared for both client and server. We can then use separate site config for client and server, which is exactly what I have done here. By doing this we can mount `document root` of client and server to separate folders. 

This will also server as an example so that anyone trying to figure out deployment will know that client and server can be in different servers and may be hosted in different servers. And they can refer to different configs as per their requirement.